### PR TITLE
update finalizer

### PR DESCRIPTION
**Pull request by huiwq1990**
use controller-runtime to delete or update finalizer.

### CODE REVIEW

File: pkg/hub/managedcluster/controller.go [hunk 1]
```diff
@@ -78,26 +79,26 @@ func (c *managedClusterController) sync(ctx context.Context, syncCtx factory.Syn
...
 	}
 
 	// Spoke cluster is deleting, we remove its related resources
 	if !managedCluster.DeletionTimestamp.IsZero() {
 		if err := c.removeManagedClusterResources(ctx, managedClusterName); err != nil {
 			return err
 		}
-		return c.removeManagedClusterFinalizer(ctx, managedCluster)
+		if controllerutil.ContainsFinalizer(managedCluster, managedClusterFinalizer) {
+			controllerutil.RemoveFinalizer(managedCluster, managedClusterFinalizer)
```
Comment (qiujian16) [inline pkg/hub/managedcluster/controller.go:96]: I hasn't take a look at the details of RemoveFinalizer, will it ensure the ordering after a deletion?

Comment (huiwq1990) [reply pkg/hub/managedcluster/controller.go:96]: Sorry, I don't understand the ordering meaing, did you say the order of finalizes? Why we need order? `RemoveFinalizer` will not change finalize order.

```
			f = append(f[:i], f[i+1:]...)

```

Comment (qiujian16) [reply pkg/hub/managedcluster/controller.go:96]: yes, the ordering of the finalizers. Some controllers may rely on the ordering of the finalizers, it is not normal but it happens.

Comment (huiwq1990) [reply pkg/hub/managedcluster/controller.go:96]: RemoveFinalizer will keep the order. By the way, could you specify which controller use this feature?

Comment (qiujian16) [reply pkg/hub/managedcluster/controller.go:96]: not in registration, but an external consumer (controller) could do that.

**Comment by qiujian16** [comment]
This may need to run `go mod tidy` &  `go mod vendor` to update the deps. The problem we met with controller-runtime is its inconsistent deps of kube related library with library-go... I think that is one of the main reason we did not use controller-runtime utility much before.

**Pull request closed without merging**